Various fixes to UI dialogs and buttons.

jcass77 opened the pull request:
In this PR:

- replace the buttons on the 'Queue' pane with jQuery Mobile's ``control group``.
- replace Javascript dialog prompts with jQuery Mobile's ``popup`` - @kingosticks, is this what you had in mind for #113?
- fixes playlist refresh button.
- update save queue functionality: now saves to 'm3u' format and no longer uses deprecated ``PlaylistsController.filter`` function.

kingosticks commented on mopidy_musicbox_webclient/static/js/controls.js:
```diff
@@ -207,37 +207,68 @@ function removeTrack() {
-    resetSong();
-    mopidy.tracklist.clear();
-    resetSong();
+    mopidy.playback.stop().then(
```
If we want we can safely omit the `stop`command since clearing does that for us.

jcass77 replied:
Ok, let's try that.

kingosticks commented on mopidy_musicbox_webclient/static/js/controls.js:
```diff
@@ -207,37 +207,68 @@ function removeTrack() {
+    mopidy.tracklist.getTracks().then(function(tracks) {
+        var plname = $('#saveinput').val().trim();
+        if (plname != null && plname != "") {
+            mopidy.playlists.filter({"name": plname}).then(function(existing) {
```
I know you are just reusing the code that was there, but we should note that `filter` is deprecated and will be removed in Mopidy v3.0

jcass77 replied:
I'll update this while I'm busy working in this area of the code.

kingosticks commented on mopidy_musicbox_webclient/static/js/controls.js:
```diff
@@ -207,37 +207,68 @@ function removeTrack() {
+    $('#popupSave').popup('close');
+    $('#saveinput').val('');
+    toast('Saving...');
+    mopidy.playlists.create({'name': playlistName, 'uri_scheme': "local"}).then(function(playlist) {
```
Again, not your doing, but using this the local uri scheme isn't right (anymore).

jcass77 replied:
On it.

kingosticks commented:
This is much smarter and defiantly addresses #113. Really nice.

Sadly, the whole queue save thing is [quite broken](https://github.com/pimusicbox/mopidy-musicbox-webclient/issues/177) and I was unable to test out the overwrite popup.

jcass77 commented:
I've pushed https://github.com/jcass77/Mopidy-MusicBox-Webclient/commit/27778751b72fbc9272e269dbb8ab22c68ad810e0 which addresses all of the comments above.

Since all of this is loosely related, I also took a stab at fixing #173 and #177 as well. The web client now listens for ``playlistChanged`` and ``playlistDeleted`` events as well, so it should rarely be necessary to use the manual refresh button. But leaving it in is probably a good idea in case there are some use cases that we have not thought of?

jcass77 commented:
https://github.com/jcass77/Mopidy-MusicBox-Webclient/commit/24100a38618e9fb18e7dd58326629afc202c70d5 changes links to proper buttons, and replaces text with equivalent FontAwesome icons to make better use of real estate on small screens.

jcass77 commented:
Rebased on develop to resolve merge conflicts.

jcass77 commented:
I can't resist the temptation of merging this and closing three issues in one fell swoop.

@kingosticks, I **think** this resolves all of the playlist-related issues - please let me know if you come across anything else that requires attention in this area.
